cli/src/profile.rs

```diff
@@ -0,0 +1,214 @@
+use addr2line::{
+    gimli::{EndianRcSlice, RunTimeEndian},
+    object::{read::File as ObjectFile, Object},
+    Context as ObjectContext, LookupResult,
+};
+use anyhow::Context;
+use probe_rs::flashing::FileDownloadError;
+use probe_rs_cli_util::{
+    clap,
+    common_options::{CargoOptions, FlashOptions, ProbeOptions},
+    flash::run_flash_download,
+};
+use std::collections::HashMap;
+use std::fs::File;
+use std::path::Path;
+use std::time::Duration;
+use strum::{Display, EnumIter, EnumString, EnumVariantNames};
+use time::Instant;
+use tracing::info;
+
+#[derive(clap::Parser)]
+pub struct ProfileOptions {
+    /// Flash the ELF before profiling
+    #[clap(long)]
+    flash: bool,
+    /// Do a full chip erase before flashing
+    #[clap(long)]
+    chip_erase: bool,
+    /// Print file and line info for each entry
+    #[clap(long)]
+    line_info: bool,
+    /// Disable double-buffering when downloading flash.  If downloading times out, try this option.
+    #[clap(long = "disable-double-buffering")]
+    disable_double_buffering: bool,
+    // Duration of profile in seconds.
+    #[clap(long)]
+    duration: u64, // Option<u64> If we could catch ctrl-c we can make this optional
```

Comment:
I think we can catch ctrl-c, no?

Reply:
Possibly, but it's not something I have time to look at now.